.github/workflows/test.build.lib.yml

```diff
@@ -1,6 +1,12 @@
 name: build:lib
 
-on: push
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
u cannot change this - there is another problem - this way you only execute the test on `main`. The purpose of the test to be always executed - the underlying problem is another - I can take a look, but I am pretty sure this is not correct.